### PR TITLE
remove internal dependency from kubectl set command

### PR DESCRIPTION
**Pull request by zjj2wry**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
ref #54212
cc @liggitt
**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by zjj2wry** [comment]
@liggitt when i remove `set subject` dependency, all version `ClusterRoleBinding` have own  version `Subject`. this not same with `v1.PodTemplate` in`workload` api.
if i want support all version, i need implement same logic in different version `ClusterRoleBinding`?

File: pkg/kubectl/cmd/set/set_selector_test.go [hunk 1]
```diff
@@ -46,14 +46,14 @@ func TestUpdateSelectorForObjectTypes(t *testing.T) {
...
-	ser := api.Service{}
-	dep := extensions.Deployment{Spec: extensions.DeploymentSpec{Selector: &before}}
-	ds := extensions.DaemonSet{Spec: extensions.DaemonSetSpec{Selector: &before}}
-	rs := extensions.ReplicaSet{Spec: extensions.ReplicaSetSpec{Selector: &before}}
-	job := batch.Job{Spec: batch.JobSpec{Selector: &before}}
-	pvc := api.PersistentVolumeClaim{Spec: api.PersistentVolumeClaimSpec{Selector: &before}}
-	sa := api.ServiceAccount{}
+	rc := v1.ReplicationController{}
+	ser := v1.Service{}
+	dep := extensionsv1beta1.Deployment{Spec: extensionsv1beta1.DeploymentSpec{Selector: &before}}
```
Comment (liggitt) [inline pkg/kubectl/cmd/set/set_selector_test.go]: for objects which exist in multiple versions, we should ensure the various versions work properly

**Comment by liggitt** [comment]
> @liggitt when i remove set subject dependency, all version ClusterRoleBinding have own version Subject. this not same with v1.PodTemplate inworkload api.
> if i want support all version, i need implement same logic in different version ClusterRoleBinding?

yes. we got lucky with podtemplate because it only exists in a single version. that's not the case for most objects

File: pkg/kubectl/cmd/set/set_selector_test.go (hunk 1 above)
Comment (zjj2wry) [reply pkg/kubectl/cmd/set/set_selector_test.go]:
```
	switch t := obj.(type) {
	case *v1.Service:
		t.Spec.Selector, err = copyOldSelector()
	default:
		err = fmt.Errorf("setting a selector is only supported for Services")
	}
```
`set selector` now just support `v1.service`.

**Comment by zjj2wry** [comment]
```
admindeMacBook-Pro-2:kubectl admin$ ./kubectl set selector -f ../../examples/guestbook/redis-master-service.yaml role=padawan
The Service "redis-master" is invalid: 
* metadata.resourceVersion: Invalid value: "": must be specified for an update
* spec.clusterIP: Invalid value: "": field is immutable

**Comment by zjj2wry** [comment]
@liggitt this ready, ptal

**Comment by zjj2wry** [comment]
@liggitt rebased, need apply LGTM once again

**Pull request merged**